src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/timeline/TimelineAdjustmentService.kt

```diff
@@ -158,6 +157,14 @@ class TimelineAdjustmentService {
     return totalAdjustmentsToApply.toLong()
   }
 
+  private fun determineSuppressSds40Hints(calculation: CalculationResult, trancheCommencementDate: LocalDate, adjustments: Adjustments): CalculationResult {
+    val suppressSds40Hints = SUPPRESS_SDS_40_HINT_ADJUSTMENT_TYPES
+      .flatMap { adjustments.getOrEmptyList(it) }
+      .any { it.appliesToSentencesFrom.isAfterOrEqualTo(trancheCommencementDate) }
+
+    return if (suppressSds40Hints) calculation.copy(showSds40Hints = false) else calculation
+  }
+
```

Comment:
The above is the main logic for the ticket, it isnt quite the same as the ticket that preceded this (CRS-2212)

This  is the description from the ticket
 suppress SDS40 default date hint text where adjustment is dated on or after prisoner’s tranche commencement date
Given dates are being calculated for a tranche 1 or tranche 2 prisoner
And the booking contains any of the following adjustments with a from date equal to or later than the prisoner’s tranche commencement date (2024-09-10 for tranche 1, 2024-10-22 for tranche 2):

Remand

Tagged bail

Unlawfully at large (UAL)

Additional days awarded (ADA)

Remission of added days (RADA)